test/test_connection.py

```diff
@@ -11,3 +13,36 @@ def test_connect_with_kwargs(self):
         client = _get_connection('foo-connection').database.client
         self.assertEqual(42, client.max_pool_size)
         self.assertEqual(10, client.min_pool_size)
+
+    def test_connect_lazily(self):
+        connect('mongodb://localhost:27017/foo',
+                'foo-connection',
+                connect=False)
+        client = _get_connection('foo-connection').database.client
+
+        class Article(MongoModel):
+            title = CharField()
+            class Meta:
+                connection_alias = 'foo-connection'
+        self.assertFalse(client._topology._opened)
+
+        self.assertEqual(Article.objects.count(), 0)
+        self.assertTrue(client._topology._opened)
+
+    def test_connect_lazily_with_index(self):
+        connect('mongodb://localhost:27017/foo',
+                'foo-connection',
+                connect=False)
+        client = _get_connection('foo-connection').database.client
+
+        class Article(MongoModel):
+            title = CharField()
+            class Meta:
+                connection_alias = 'foo-connection'
+                indexes = [
+                    IndexModel([('title', 1)])
+                ]
+        self.assertFalse(client._topology._opened)
+
+        self.assertEqual(Article.objects.count(), 0)
+        self.assertTrue(client._topology._opened)
```

Comment:
Same comment here. We should also add a createIndexes command listener to make sure the createIndexes is run:

```python
from pymongo.monitoring import ServerHeartbeatListener

class WhiteListEventListener(monitoring.CommandListener):
    def __init__(self, *commands):
        self.commands = set(commands)
        self.results = defaultdict(list)

    def started(self, event):
        if event.command_name in self.commands:
            self.results['started'].append(event)

    def succeeded(self, event):
        if event.command_name in self.commands:
            self.results['succeeded'].append(event)

    def failed(self, event):
        if event.command_name in self.commands:
            self.results['failed'].append(event)

heartbeat_listener = HeartbeatStartedListener()
create_indexes_listener = WhiteListEventListener("createIndexes")
connect('mongodb://localhost:27017/foo',
        'foo-connection',
         connect=False,
         event_listeners=[heartbeat_listener, create_indexes_listener])
...

self.assertEqual(len(heartbeat_listener.results), 0)
self.assertEqual(len(create_indexes_listener.results['started']), 0)
self.assertEqual(Article.objects.count(), 0)
self.assertGreaterOrEqual(len(heartbeat_listener.results), 1)
self.assertEqual(len(create_indexes_listener.results['started']), 1)
```